aten/src/ATen/native/Math.h

```diff
@@ -8,9 +8,12 @@
 #include <type_traits>
 #include <c10/util/math_compat.h>
 
-#ifndef M_PIf
-#define M_PIf 3.1415926535f
-#endif  // M_PIf
+namespace at {
+namespace native {
+//TODO: Replace me with std::numbers::pi when C++20 becomes available
+template <typename T> constexpr T pi = 3.14159265358979323846L;
```

Comment:
Do we know the impact of replacing `M_PI` with this value? Could that lead to some precision changes?

Reply:
To ensure there are none, I can add .cpp file that include <math.h> and adds `static_assert(at:native::pi<double> == M_PI)`
The motivation is not to pollute global namespace with M_PI define and also prevent build system brittleness on Windows, where including <cmath> before <math.h> would result in M_PI not being defined

Reply:
That sounds good.

> To ensure there are none, I can add .cpp file that include <math.h> and adds static_assert(at:native::pi<double> == M_PI)

Would that be brittle on windows? haha 
It would be nice though to have a dummy PR that adds this check and makes sure it passes on all our CI just as a sanity check.